--- httpd-2.4.39.orig/modules/ssl/ssl_engine_vars.c	2019-04-29 00:30:14.000000000 +0200
+++ httpd-2.4.39/modules/ssl/ssl_engine_vars.c	2019-04-29 00:29:48.000000000 +0200
@@ -68,7 +68,7 @@ static SSLConnRec *ssl_get_effective_con
 static int ssl_is_https(conn_rec *c)
 {
     SSLConnRec *sslconn = ssl_get_effective_config(c);
-    return sslconn && sslconn->ssl;
+    return sslconn && sslconn->ssl || (apr_table_get(c->notes, "remoteip-https") != NULL);
 }
 
 static const char var_interface[] = "mod_ssl/" AP_SERVER_BASEREVISION;
